com.unity.ml-agents.extensions/Runtime/Match3/AbstractBoard.cs

```diff
@@ -86,45 +115,37 @@ public int NumMoves()
         /// <returns></returns>
         public IEnumerable<Move> AllMoves()
         {
-            var currentMove = Move.FromMoveIndex(0, Rows, Columns);
-            for (var i = 0; i < NumMoves(); i++)
-            {
-                yield return currentMove;
-                currentMove.Next(Rows, Columns);
-            }
-        }
+            var maxBoardSize = GetMaxBoardSize();
+            var currentBoardSize = GetCurrentBoardSize();
 
-        /// <summary>
-        /// Iterate through all valid Moves on the board.
-        /// </summary>
-        /// <returns></returns>
-        public IEnumerable<Move> ValidMoves()
-        {
-            var currentMove = Move.FromMoveIndex(0, Rows, Columns);
+            var currentMove = Move.FromMoveIndex(0, maxBoardSize.Rows, maxBoardSize.Columns);
             for (var i = 0; i < NumMoves(); i++)
             {
-                if (IsMoveValid(currentMove))
+                if (currentMove.IsValidForBoardSize(currentBoardSize))
                 {
                     yield return currentMove;
                 }
-                currentMove.Next(Rows, Columns);
+                currentMove.Next(maxBoardSize.Rows, maxBoardSize.Columns);
             }
         }
 
         /// <summary>
-        /// Iterate through all invalid Moves on the board.
+        /// Iterate through all valid Moves on the board.
         /// </summary>
         /// <returns></returns>
-        public IEnumerable<Move> InvalidMoves()
```

Comment:
The diff here is messy. InvalidMoves() was removed (previously it was used for action masking, but not anymore.